cmd/server/pactasrv/blobs.go

```diff
@@ -0,0 +1,112 @@
+package pactasrv
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/RMI/pacta/db"
+	"github.com/RMI/pacta/oapierr"
+	api "github.com/RMI/pacta/openapi/pacta"
+	"github.com/RMI/pacta/pacta"
+	"go.uber.org/zap"
+)
+
+func (s *Server) AccessBlobContent(ctx context.Context, request api.AccessBlobContentRequestObject) (api.AccessBlobContentResponseObject, error) {
+	actorOwnerID, err := s.getUserOwnerID(ctx)
+	if err != nil {
+		return nil, err
+	}
+	actorUserID, err := getUserID(ctx)
+	if err != nil {
+		return nil, err
+	}
+	actorIsAdmin, actorIsSuperAdmin, err := s.isAdminOrSuperAdmin(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	blobIDs := []pacta.BlobID{}
+	for _, item := range request.Body.Items {
+		blobIDs = append(blobIDs, pacta.BlobID(item.BlobId))
+	}
+	err404 := oapierr.NotFound("blob not found", zap.Strings("blob_ids", asStrs(blobIDs)))
+	bois, err := s.DB.BlobOwners(s.DB.NoTxn(ctx), blobIDs)
+	if err != nil {
+		if db.IsNotFound(err) {
+			return nil, err404
+		}
+		return nil, oapierr.Internal("error getting blob owners", zap.Error(err), zap.Strings("blob_ids", asStrs(blobIDs)))
+	}
+	asMap := map[pacta.BlobID]*pacta.BlobOwnerInformation{}
+	for _, boi := range bois {
+		asMap[boi.BlobID] = boi
+	}
+	auditLogs := []*pacta.AuditLog{}
+	for _, blobID := range blobIDs {
+		boi := asMap[blobID]
+		accessAsOwner := boi.PrimaryTargetOwnerID == actorOwnerID
+		accessAsAdmin := boi.AdminDebugEnabled && actorIsAdmin
+		accessAsSuperAdmin := boi.AdminDebugEnabled && actorIsSuperAdmin
+		al := &pacta.AuditLog{
+			Action:             pacta.AuditLogAction_Download,
+			ActorID:            string(actorUserID),
+			ActorOwner:         &pacta.Owner{ID: actorOwnerID},
+			PrimaryTargetType:  boi.PrimaryTargetType,
+			PrimaryTargetID:    boi.PrimaryTargetID,
+			PrimaryTargetOwner: &pacta.Owner{ID: boi.PrimaryTargetOwnerID},
+		}
+		if accessAsOwner {
+			al.ActorType = pacta.AuditLogActorType_User
+		} else if accessAsAdmin {
+			al.ActorType = pacta.AuditLogActorType_Admin
+		} else if accessAsSuperAdmin {
+			al.ActorType = pacta.AuditLogActorType_SuperAdmin
+		} else {
+			// DENY CASE
+			return nil, err404
+		}
+		auditLogs = append(auditLogs, al)
```

Comment:
optional: I find code like this easier to understand and modify when the object is constructed once and not changed, e.g.:

```go
var actorType pacta.AuditLogActorType
if accessAsOwner {
	actorType = pacta.AuditLogActorType_User
} else if accessAsAdmin {
	actorType = pacta.AuditLogActorType_Admin
} else if accessAsSuperAdmin {
	actorType = pacta.AuditLogActorType_SuperAdmin
} else {
	// DENY CASE
	return nil, err404
}
auditLogs = append(auditLogs, &pacta.AuditLog{
	Action:             pacta.AuditLogAction_Download,
	ActorType:          actorType,
	ActorID:            string(actorUserID),
	ActorOwner:         &pacta.Owner{ID: actorOwnerID},
	PrimaryTargetType:  boi.PrimaryTargetType,
	PrimaryTargetID:    boi.PrimaryTargetID,
	PrimaryTargetOwner: &pacta.Owner{ID: boi.PrimaryTargetOwnerID},
})
```

I only make this (otherwise pretty pedantic) note because it sounds like this is a pretty sensitive code path.

Reply:
SGTM